org.code.toboggan.ui/src/org/code/toboggan/ui/error/response/FileResponseErrorDisplay.java

```diff
@@ -0,0 +1,79 @@
+package org.code.toboggan.ui.error.response;
+
+import java.nio.file.Path;
+
+import org.code.toboggan.network.request.extensionpoints.file.IFileChangeResponse;
+import org.code.toboggan.network.request.extensionpoints.file.IFileCreateResponse;
+import org.code.toboggan.network.request.extensionpoints.file.IFileDeleteResponse;
+import org.code.toboggan.network.request.extensionpoints.file.IFileMoveResponse;
+import org.code.toboggan.network.request.extensionpoints.file.IFilePullDiffSendChangesResponse;
+import org.code.toboggan.network.request.extensionpoints.file.IFilePullResponse;
+import org.code.toboggan.network.request.extensionpoints.file.IFileRenameResponse;
+import org.code.toboggan.ui.dialogs.MessageDialog;
+
+import clientcore.patching.Patch;
+
+public class FileResponseErrorDisplay implements IFileChangeResponse, IFileCreateResponse, IFileDeleteResponse,
+		IFileMoveResponse, IFilePullDiffSendChangesResponse, IFilePullResponse, IFileRenameResponse {
+
+	@Override
+	public void fileRenamed(long fileID, Path newFileLocation, String newName) {
+		// Do nothing
+	}
+
+	@Override
+	public void fileRenameFailed(long fileID, Path oldFileLocation, Path newFileLocation, String newName) {
+		MessageDialog.createDialog("Failed to rename the file on the server. Please resubscribe to the project.").open();
+	}
+
+	@Override
+	public void filePulled(long fileID, byte[] fileBytes, String[] changes) {
+		// Do nothing
+	}
+
+	@Override
+	public void filePullFailed(long fileID) {
+		MessageDialog.createDialog("Failed to pull file from the server. Please resubscribe to the project.").open();
+	}
+
+	@Override
+	public void fileMoved(long fileID, Path newWorkspaceRelativePath) {
```

Comment:
Might be a good idea to change these parameter names to say `absolutePath` or `fileLocation` so as to not confuse anyone. I've mainly been using the latter since it is more explicit and aligns with how Eclipse denotes paths on disk.